File: src/ch07/ex10/ex10.java
package ch07.ex10;

import java.util.Random;

import javafx.application.Application;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.scene.Group;
import javafx.scene.Scene;
import javafx.scene.chart.PieChart;
import javafx.stage.Stage;

public class ex10 extends Application {
	public static void main(String[] args) {
		launch();
	}

	@Override
	public void start(Stage stage) throws Exception {
		Random rnd = new Random();

		ObservableList<PieChart.Data> pieChartData = FXCollections.observableArrayList(
				new PieChart.Data("A3横", rnd.nextInt(100)), new PieChart.Data("A3縦", rnd.nextInt(100)),
				new PieChart.Data("A4横", rnd.nextInt(100)), new PieChart.Data("A4縦", rnd.nextInt(100)),
				new PieChart.Data("B4横", rnd.nextInt(100)), new PieChart.Data("B4縦", rnd.nextInt(100)));
		PieChart chart = new PieChart(pieChartData);

		Scene scene = new Scene(new Group());
		((Group) scene.getRoot()).getChildren().add(chart);
		stage.setWidth(500);
		stage.setHeight(500);
		stage.setScene(scene);
		stage.show();
	}
}

File: src/ch07/ex10/ex10.js
var StandardCharsets = java.nio.charset.StandardCharsets;
var Files = java.nio.file.Files;
var Paths = java.nio.file.Paths;
var String = java.lang.String;
var FXCollections = javafx.collections.FXCollections;
var PieChart = javafx.scene.chart.PieChart;

var path = "C:\\a\\pleiades\\workspace\\Java\\src\\ch07\\ex10\\ex10.json";
var contents = new String(Files.readAllBytes(Paths.get(path)), StandardCharsets.UTF_8);
var json = JSON.parse(contents);

var pieChartData = FXCollections.observableArrayList();
for (var n in json.data) {
  pieChartData.add(new PieChart.Data(json.data[n].type, +json.data[n].papers));
}
var chart = new PieChart(pieChartData);

$STAGE.scene = new javafx.scene.Scene(chart, 600, 400)